lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -34,30 +42,51 @@
 
 
 @attr('shard_1')
+@ddt.ddt
 class AccessTestCase(LoginEnrollmentTestCase, ModuleStoreTestCase):
     """
     Tests for the various access controls on the student dashboard
     """
+    TOMORROW = datetime.datetime.now(pytz.utc) + datetime.timedelta(days=1)
+    ADVERTISED_START = "Spring 2016"
+
     def setUp(self):
         super(AccessTestCase, self).setUp()
         course_key = SlashSeparatedCourseKey('edX', 'toy', '2012_Fall')
         self.course = course_key.make_usage_key('course', course_key.run)
         self.anonymous_user = AnonymousUserFactory()
+        self.beta_user = BetaTesterFactory(course_key=self.course.course_key)
         self.student = UserFactory()
         self.global_staff = UserFactory(is_staff=True)
         self.course_staff = StaffFactory(course_key=self.course.course_key)
         self.course_instructor = InstructorFactory(course_key=self.course.course_key)
+        self.staff = Mock(is_staff=True)
 
     def verify_access(self, mock_unit, student_should_have_access):
         """ Verify the expected result from _has_access_descriptor """
+
         self.assertEqual(
             student_should_have_access,
-            access._has_access_descriptor(self.anonymous_user, 'load', mock_unit, course_key=self.course.course_key)
+            bool(access._has_access_descriptor(
+                self.anonymous_user, 'load', mock_unit, course_key=self.course.course_key))
         )
+
         self.assertTrue(
             access._has_access_descriptor(self.course_staff, 'load', mock_unit, course_key=self.course.course_key)
         )
 
+    def verify_error_type(self, mock_unit, expected_error_type):
```

Comment:
Instead of introducing a new `verify_error_type` function, update the existing `verify_access` function to take in the expected response type.